File: OOP/typescript/Inheritance/hybrid.ts
 // Base Interface
interface UniversityMember {
  showID(): void;
}

// First Branch
interface Staff extends UniversityMember {
  showSalary(): void;
}

// Second Branch
interface Student extends UniversityMember {
  showCredits(): void;
}

// Class implementing multiple interfaces
class GraduateInstructor implements Staff, Student {
  showID() {
    console.log("ID: U12345");
  }

  showSalary() {
    console.log("Salary: ₹25,000");
  }

  showCredits() {
    console.log("Credits: 12");
  }
}

const gi = new GraduateInstructor();
gi.showID();
gi.showSalary();
gi.showCredits();

File: OOP/java/Inheritance/Hybrid.java
package Inheritance;

// Base Interface
interface UniversityMember {
    void showID();
}

// First Branch
interface Staff extends UniversityMember {
    void showSalary();
}

// Second Branch
interface Student extends UniversityMember {
    void showCredits();
}

// Multi-inheriting Interface
class GraduateInstructor implements Staff, Student {

    public void showID() {
        System.out.println("ID: U12345");
    }

    public void showSalary() {
        System.out.println("Salary: ₹25,000");
    }

    public void showCredits() {
        System.out.println("Credits: 12");
    }
}

public class Hybrid {
    public static void main(String[] args) {
        GraduateInstructor gi = new GraduateInstructor();
        gi.showID();
        gi.showSalary();
        gi.showCredits();
    }
}

